components/locale/src/extensions/private/key.rs

```diff
@@ -0,0 +1,73 @@
+use std::ops::RangeInclusive;
+use std::str::FromStr;
+
+use crate::parser::errors::ParserError;
+use tinystr::TinyStr8;
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash, PartialOrd, Ord, Copy)]
+pub struct Key(TinyStr8);
+
+const KEY_LENGTH: RangeInclusive<usize> = 1..=8;
+
+impl Key {
+    /// A constructor which takes a utf8 slice, parses it and
+    /// produces a well-formed `Variant`.
```

Comment:
Using 'Variant' here is maybe too overloaded. Maybe opaque value?

Reply:
it's a copy&paste from Variant. Fixed!